code/modules/mob/abstract/new_player/new_player.dm

```diff
@@ -133,6 +133,22 @@ INITIALIZE_IMMEDIATE(/mob/abstract/new_player)
 			to_chat(usr, "<span class='danger'>Your current species, [client.prefs.species], is not available for play on the station.</span>")
 			return 0
 
+		var/confirm_job = FALSE
+		var/datum/job/J = SSjobs.GetJob(href_list["SelectedJob"])
+		switch(J.department_flag)
+			if(CIVILIAN)
+				if(!((client.prefs.job_civilian_high || client.prefs.job_civilian_high || client.prefs.job_civilian_high) == J.flag))
```

Comment:
i don't think the code functions as intended here, you're gonna want to do something like:
`if(client.prefs.job_civilian_high != J.flag && client.prefs.job_civilian_high != J.flag && client.prefs.job_civilian_high != J.flag))`
for all of them

Reply:
I tested it with this, and it works in exactly the same way

Reply:
Tested it locally. Currently it checks the highest preference set for the category, and only doesn't ask if you got a single of that. In essence:
-If you got a highest, it will not ask for highest but for all other med & low jobs.
-If you got two medium but no highest, it will ask no matter which of the medium is clicked.
-If you got a single medium and no highest, it will not ask when clicking the medium.

I also tested the bit Geeves gave and it meets the same problem if there are two jobs in the same category set to medium.